File: library/src/main/java/me/proxer/library/entity/user/UserAbout.java
package me.proxer.library.entity.user;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Value;
import me.proxer.library.enums.Gender;
import me.proxer.library.enums.RelationshipStatus;

import javax.annotation.Nullable;
import java.util.Date;

/**
 * Entity holding all info of a user.
 *
 * @author Ruben Gees
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class UserAbout {

    /**
     * Returns the website of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_website")
    private String website;

    /**
     * Returns the occupation of the user. Can be empty.
     */
    @Json(name = "info_occupation")
    private String occupation;

    /**
     * Returns the interests of the user. Can be empty.
     */
    @Json(name = "info_interests")
    private String interests;

    /**
     * Returns the city of the user. Can be empty.
     */
    @Json(name = "info_city")
    private String city;

    /**
     * Returns the country of the user. Can be empty.
     */
    @Json(name = "info_country")
    private String country;

    /**
     * Returns the about text of the user. Can be empty and can contain HTML-Tags.
     */
    @Json(name = "info_about")
    private String about;

    /**
     * Returns the facebook info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_facebook")
    private String facebook;

    /**
     * Returns the youtube info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_youtube")
    private String youtube;

    /**
     * Returns the chatango info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_chatango")
    private String chatango;

    /**
     * Returns the twitter info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_twitter")
    private String twitter;

    /**
     * Returns the skype info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_skype")
    private String skype;

    /**
     * Returns the deviantart info of the user. Can be empty, a link or a name.
     */
    @Json(name = "info_deviantart")
    private String deviantart;

    /**
     * Returns the birthday of the user. Can be an actual date or 00.00.0000. The format is dd.MM.yyyy.
     */
    @Json(name = "info_birthday")
    private Date birthday;

    /**
     * Returns the gender of the user.
     */
    @Json(name = "info_gender")
    private Gender gender;

    /**
     * Returns the relationship status of the user.
     */
    @Json(name = "info_relationshipstatus")
    private RelationshipStatus relationshipStatus;
}

File: library/src/main/kotlin/me/proxer/library/entity/user/UserAbout.kt
package me.proxer.library.entity.user

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.enums.Gender
import me.proxer.library.enums.RelationshipStatus

/**
 * Entity holding all info of a user.
 *
 * @property website The website of the user. Can be empty, a link or a name.
 * @property occupation The occupation of the user. Can be empty.
 * @property interests The interests of the user. Can be empty.
 * @property city The city of the user. Can be empty.
 * @property country The country of the user. Can be empty.
 * @property about The about text of the user. Can be empty and can contain HTML-Tags.
 * @property facebook The facebook info of the user. Can be empty, a link or a name.
 * @property youtube The youtube info of the user. Can be empty, a link or a name.
 * @property chatango The chatango info of the user. Can be empty, a link or a name.
 * @property twitter The twitter info of the user. Can be empty, a link or a name.
 * @property skype The skype info of the user. Can be empty, a link or a name.
 * @property deviantart The deviantart info of the user. Can be empty, a link or a name.
 * @property birthday The birthday of the user. Can be an actual date or 00.00.0000. The format is dd.MM.yyyy.
 * @property gender The gender of the user.
 * @property gender The relationship status of the user.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class UserAbout(
    @Json(name = "info_website") val website: String,
    @Json(name = "info_occupation") val occupation: String,
    @Json(name = "info_interests") val interests: String,
    @Json(name = "info_city") val city: String,
    @Json(name = "info_country") val country: String,
    @Json(name = "info_about") val about: String,
    @Json(name = "info_facebook") val facebook: String,
    @Json(name = "info_youtube") val youtube: String,
    @Json(name = "info_chatango") val chatango: String,
    @Json(name = "info_twitter") val twitter: String,
    @Json(name = "info_skype") val skype: String,
    @Json(name = "info_deviantart") val deviantart: String,
    @Json(name = "info_birthday") val birthday: String,
    @Json(name = "info_gender") val gender: Gender,
    @Json(name = "info_relationshipstatus") val relationshipStatus: RelationshipStatus
)
